### PR TITLE
tests/arduino: add automatic test for arduino module

### PR DESCRIPTION
**Pull request by fedepell**
### Contribution description
This PR adds an automatic test for the arduino module using the standard RIOT testing facilities. 
The goal is to provide an easier way to test the module when we apply changes and maybe even integrate it into the CI system.
The current code will test:
- if code is built correctly (sketch conversion)
- basic serial read/write functionality
- serial printing routines for various formats conversion (dec,hex,oct)
- time printing routines and sleep verification

I tested this on my saml21-xpro.

### Testing procedure
A board with arduino support is needed (arduino_board.h present), then in the tests/sys_arduino:

`BOARD=<board name> make all flash test`

This will build the test, flash it and execute the test.

### CODE REVIEW

File: tests/sys_arduino/arduino-test.sketch [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+void loop(void)
+{
+    /* Read chars if available and seek for CR or LF */
+    while (Serial.available() > 0) {
+        int tmp = Serial.read();
+        if (tmp == '\n' || tmp == '\r') {
+           buf[count] = 0;
+
+           if (count > 1) {
+                if (strncmp(buf, "echo", 4) == 0) {
```
Comment (aabadie) [inline tests/sys_arduino/arduino-test.sketch]: You are not testing this command in the Python script.

Comment (fedepell) [reply tests/sys_arduino/arduino-test.sketch]: Thanks for checking out and for spotting I forgot the first and most basic test I had in mind to insert :) 

I added now a test to use that echoing command and squashed up the changes.

File: tests/sys_arduino/arduino-test.sketch [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                    /* echo command just echoes back the input string */
+                    Serial.write("ECHO:");
+                    for (int i = 4; i < count; i++) {
+                        Serial.write(buf[i]);
+                    }
+                } else if (strncmp(buf, "numb", 4) == 0) {
+                    /* numb command prints input number if various formats */
```
Comment (aabadie) [inline tests/sys_arduino/arduino-test.sketch]: typo: `if various formats` should be `in various formats`

Comment (fedepell) [reply tests/sys_arduino/arduino-test.sketch]: Thanks for spotting this! Fixed!

**Comment by fedepell** [comment]
I checked the failure in Murdock but seems unrelated to the changes, correct? Like some hardware/flasher was not available?

**Comment by miri64** [comment]
Yepp, there is some missconfigured Pi within the testing Pi-fleet. Hopefully, this will be fixed soon.

**Comment by fedepell** [comment]
> Yepp, there is some missconfigured Pi within the testing Pi-fleet. Hopefully, this will be fixed soon.

Many thanks for the prompt clarification!

**Comment by aabadie** [comment]
Since this PR is adding a runnable test, I triggered the `CI: run tests` label but it was not a good idea. Anyway, I tested this PR locally on Arduino Zero and the test have passed with success. Let's wait for Murdock and merge.

**Comment by aabadie** [comment]
All green now, let's go!

**Pull request merged**